File: ChineseAuctionAPI/ChineseAuctionAPI/DTOs/UserDTO.cs
using System.ComponentModel.DataAnnotations;
using Swashbuckle.AspNetCore.SwaggerGen;
using DataType = System.ComponentModel.DataAnnotations.DataType;

namespace ChineseAuctionAPI.DTOs
{
    public class UserDTO
    {
        [Required]
        [Phone]
        public string PhoneNumber { get; set; }
        [Required]
        public string FirstName { get; set; }
        [Required]
        public string LastName { get; set; }
        public string City { get; set; }
        public string Address { get; set; }
        [EmailAddress(ErrorMessage = "כתובת האימייל אינה תקינה")]
        public string? Email { get; set; }
        [Required]
        [RegularExpression(@"^\d{9}$", ErrorMessage = "תעודת זהות חייבת להכיל 9 ספרות")]
        public string Identity { get; internal set; }

    }
    public class DtoLogin
    {
        [Required,MinLength(6)]
        [DataType(DataType.Password)]
        public string Password { get; set; }
        [Required]
        [RegularExpression(@"^\d{9}$", ErrorMessage = "תעודת זהות חייבת להכיל 9 ספרות")]
        public string Identity { get; set; }
        [Required]
        public string FirstName { get; set; }
        [Required]
        public string LastName { get; set; }
        [EmailAddress(ErrorMessage = "כתובת האימייל אינה תקינה")]
        public string? Email { get; set; }
        [Phone,Required]
        public string PhoneNumber { get; set; }
        public string City { get; set; }
        public string Address { get; set; }
    }

    public class DtoUserOrder
    {
        [Required]
        public string FirstName { get; set; }
        [Required]
        public string LastName { get; set; }
        public List<OrderDTO> Orders { get; set; } = new();
    }
    public class DtoLoginResponse
    {
        [Required]
        public string Token { get; set; }
        [Required]
        public UserDTO User { get; set; }
    }
    public class DtologinRequest
    {
        [EmailAddress,Required]
        public string Email { get; set; }
        [Required,MinLength(6), DataType(DataType.Password)]
        public string Password { get; set; }
    }
}

File: ChineseAuctionAngular/ChineseAuctionAngular/src/app/models/UserDTO.ts
import { OrderDTO } from "./OrderDTO";

// 1. UserDTO Class
export class UserDTO {
  phoneNumber: string = '';
  firstName: string = '';
  lastName: string = '';
  city?: string = '';
  address?: string = '';
  email?: string = '';
  identity: string = '';

  constructor(init?: Partial<UserDTO>) {
    Object.assign(this, init);
  }
}

export class DtoLogin {
  password?: string = '';
  identity: string = '';
  firstName: string = '';
  lastName: string = '';
  email?: string = '';
  phoneNumber: string = '';
  city: string = '';
  address: string = '';

  constructor(init?: Partial<DtoLogin>) {
    Object.assign(this, init);
  }
}

// 3. DtologinRequest Class (עבור מסך התחברות)
export class DtoLoginRequest {
  email: string = '';
  password: string = '';

  constructor(init?: Partial<DtoLoginRequest>) {
    Object.assign(this, init);
  }
}

// 4. DtoLoginResponse Class
export class DtoLoginResponse {
  token: string = '';
  user: UserDTO = new UserDTO();

  constructor(init?: Partial<DtoLoginResponse>) {
    Object.assign(this, init);
  }
}

// 5. DtoUserOrder Class
export class DtoUserOrder {
  firstName: string = '';
  lastName: string = '';
  orders: OrderDTO[] = []; // הוחלף מ-any[]

  constructor(init?: Partial<DtoUserOrder>) {
    Object.assign(this, init);
  }
}